cloudslang-enforcer/src/main/java/io/cloudslang/lang/enforcer/SpringCleansingRule.java

```diff
@@ -0,0 +1,198 @@
+            // RuntimeInfo rti = (RuntimeInfo) helper.getComponent(RuntimeInfo.class);
+            List compileSourceRoots = project.getCompileSourceRoots();
+
+            for (Object compileSourceRoot : compileSourceRoots) {
+                String path = (String) compileSourceRoot;
+                applyForJavaSourcesInRoot(path, log);
+            }
+        } catch (ExpressionEvaluationException e) {
+            throw new EnforcerRuleException("Unable to lookup an expression " + e.getLocalizedMessage(), e);
+        }
+    }
+
+    /**
+     * If your rule is cacheable, you must return a unique id when parameters or conditions
+     * change that would cause the result to be different. Multiple cached results are stored
+     * based on their id.
+     * <p>
+     * The easiest way to do this is to return a hash computed from the values of your parameters.
+     * <p>
+     * If your rule is not cacheable, then the result here is not important, you may return anything.
+     */
+    public String getCacheId() {
+        //no hash on boolean...only parameter so no hash is needed.
+        return "" + this.shouldIfail;
+    }
+
+    /**
+     * This tells the system if the results are cacheable at all. Keep in mind that during
+     * forked builds and other things, a given rule may be executed more than once for the same
+     * project. This means that even things that change from project to project may still
+     * be cacheable in certain instances.
+     */
+    public boolean isCacheable() {
+        return false;
+    }
+
+    /**
+     * If the rule is cacheable and the same id is found in the cache, the stored results
+     * are passed to this method to allow double checking of the results. Most of the time
+     * this can be done by generating unique ids, but sometimes the results of objects returned
+     * by the helper need to be queried. You may for example, store certain objects in your rule
+     * and then query them later.
+     */
+    public boolean isResultValid(EnforcerRule arg0) {
+        return false;
+    }
+
+    private boolean isSpringConfigurationAnnotatedClass(final String contents) {
+        Matcher matcher = patternConfigurationClass.matcher(contents);
+        return matcher.find();
+    }
+
+    private void applyForJavaSourcesInRoot(final String path, final Log log) throws EnforcerRuleException {
+        Iterator<File> fileIterator = iterateFiles(new File(path), new String[]{JAVA}, true);
+        while (fileIterator.hasNext()) {
+            File source = fileIterator.next();
+
+            if (isRegularFile(source.toPath(), NOFOLLOW_LINKS)) {
+                if (log.isDebugEnabled()) {
+                    log.debug(format(STARTED_SCANNING_ORG_SPRINGFRAMEWORK, source.getAbsolutePath()));
+                }
+                try {
+                    String contents = readFileToString(source, UTF_8.displayName());
```

Comment:
you may consider using CS charset her ~ SlangSource.getCloudSlangCharset() (we could extract this method in `commons`)